### PR TITLE
添加新文章「在 Fedora 上入门 Jupyter 与数据科学」

### PR DESCRIPTION
**Pull request by geekpany**
Signed-off-by: Pany <pany@fedoraproject.org>

### CODE REVIEW

File: _posts/2019-07-03-jupyter-and-data-science-in-fedora.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+---
+layout: post
+title: 在 Fedora 上入门 Jupyter 与数据科学
+date: 2019-07-24 22:39:38 +0800
+---
+
+> 本文翻译自 Avi Alkalay 的文章 [Jupyter and data science in Fedora](https://fedoramagazine.org/jupyter-and-data-science-in-fedora/)。
+
+在过去，君主和领袖会依赖先知和巫师来预测未来——或者借助其神秘力量来探测潜在信息。现如今，我们生活在一个格物致知的世界之中。于是我们便有了数据科学来完成这些工作。
```
Comment (sztsian) [inline _posts/2019-07-03-jupyter-and-data-science-in-fedora.md]: `格物致知` 我倾向于直译。。

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
+---
+layout: post
+title: 在 Fedora 上入门 Jupyter 与数据科学
+date: 2019-07-24 22:39:38 +0800
+---
+
+> 本文翻译自 Avi Alkalay 的文章 [Jupyter and data science in Fedora](https://fedoramagazine.org/jupyter-and-data-science-in-fedora/)。
+
+在过去，君主和领袖会依赖先知和巫师来预测未来——或者借助其神秘力量来探测潜在信息。现如今，我们生活在一个格物致知的世界之中。于是我们便有了数据科学来完成这些工作。
```
Comment (sztsian) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: `Nowadays, we live in a society obsessed with quantifying everything.`  建议直译。感觉 `格物致知` 在这儿虽然读起来很文雅，但不利于理解背景。

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+数据科学家使用统计学科之外的统计模型、数学技术和高级算法，针对数据库中存在的数据，进行查找、推断、预测潜在的隐性数据。有时，这些数据就关乎未来。正因如此，我们才会去做很多预测性分析和规范性分析。
+
+数据科学家会致力于研究类似以下这些问题：
+
+1. 哪些学生比较有翘课倾向？具体到个人，其背后的原因有哪些？
+1. 哪套房子的价格高于或低于均价？对于某一套房，其合理的价格是多少？
```
Comment (sztsian) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: `哪套房子的价格高于或低于均价？` 我觉得这儿不宜翻译为均价。 Fair price 理解为合理的价位更适合这个场景。例如 `哪套房子价格超出合理或者低于合理价位`

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+要知道以上这些问题的答案并不会躺在哪个数据库里等人去查。这些数据都是需要计算的，不然根本就不会存在。而这正是数据科学家工作内容的一部分。
+
+在本文中，你可以学到如何将 Fedora 系统打造成为数据科学家的开发及生产系统。大部分基础软件都打好 RPM 包了，不过目前比较高阶的部分只能通过 Python 的 *pip* 工具安装。
+
+## Jupyter——集成开发环境
+
+大部分新潮的数据科学家都用 Python。而他们工作中非常重要的一环就是 EDA（探索性数据分析）。EDA 是一个手动和交互式过程，可以检索数据，探索其特性，搜索相关性，并通过绘制图形来进行可视化，进而理解数据的全貌和原型预测模型。
+
+Jupyter 是一款可以完美适用于这些任务的 web 应用。Jupyter 可以搭配笔记本（Notebooks）、文档来使用，其中可以整合富文本，包括渲染精美的数学公式（得益于 [mathjax](http://mathjax.org) 的加持）、代码块、代码输出，甚至还包括输出的图像等。
```
Comment (sztsian) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: Notebooks 作为专有名词，不翻译也许更好。下同。

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+     "open_browser": false,
+     "websocket_compression_options": {},
+     "certfile": "/home/aviram/.jupyter/service.pem",
+     "keyfile": "/home/aviram/.jupyter/service-key.pem",
+     "notebook_dir": "/home/aviram/Notebooks"
+   }
+}
+```
+
+涉及到文件路径时，用户名的部分需要改成你自己的。密码那一串是当你创建密码时自动生成的。pem 相关的那一部分是 *sscg* 创建的加密相关的文件。
```
Comment (sztsian) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: `用户名的部分` 建议直接写成 `/home/aviram 部分`，或者 `家目录的路径`。主要是有些场景（例如我的某些工作环境），我的家目录并不在 /home/zsun

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+![利用 SeaBorn 进行线性回归的可视化](/assets/2019/07/24/regression_marginals.png)
+
+*利用 SeaBorn 进行线性回归的可视化*
+
+### [StatsModels](https://www.statsmodels.org/)
+
+StatsModels 可以为统计和计量经济学数据分析提供算法，如线性和逻辑回归。Statsmodel 同样适用于 ARIMA 等经典[时间序列算法](https://www.statsmodels.org/stable/examples/index.html#stats)系列。
+
+![标准化之后的旅客量随时间变化趋势（蓝线）与 ARIMA 预测的旅客量（红线）](/assets/2019/07/24/time-series.png)
```
Comment (sztsian) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: 我们上学的时候 normalized 中文用的是 `归一化`。也许得看下 @alick 用的什么。

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+*标准化之后的旅客量随时间变化趋势（蓝线）与 ARIMA 预测的旅客量（红线）*
+
+### Scikit-learn
+
+作为机器学习生态系统的核心部分，[scikit](https://scikit-learn.org/stable/) 为[回归](https://scikit-learn.org/stable/supervised_learning.html#supervised-learning)（弹性网、梯度提升、随机森林等）以及[分类](https://scikit-learn.org/stable/supervised_learning.html#supervised-learning)和聚类（K-means、DBSCAN等）提供了预测算法。它的一大特性就是 API 设计得非常好。Scikit 有很多类别，可用于高级数据操作、将数据集拆分为训练和测试部分、降维以及数据管道准备等。
```
Comment (sztsian) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: `提供了预测算法`  ->  `提供了算法`

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
+[imbalanced-learn](https://imbalanced-learn.readthedocs.io) 可以提供对数据进行欠采样和过采样的方法。这在欺诈检测场景中非常实用，因为与非欺诈数据相比，已知的欺诈数据会非常小。这时，就需要针对已知欺诈数据进行数据扩大，以使其与训练预测工具更相匹配。使用 *pip* 安装：
+
+```
+$ pip3 install imblearn --user
+```
+
+### NLTK
+
+[自然语言工具箱](https://www.nltk.org)，或者叫 NLTK，可以帮我们处理人类语言数据，从而实现构建聊天机器人等工具（只是举个例子）。
```
Comment (sztsian) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: `从而实现构建聊天机器人等工具（只是举个例子）` -> `从而实现构建例如聊天机器人之类的工具`

**Review by sztsian — CHANGES_REQUESTED** [review]
一些建议。

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+在笔者看来，Python 之所以能成为数据科学的热门选择，Pandas 功不可没。Pandas 基于 numpy，它可以让准备和显示数据的工作变得更加轻松。你可以把它想象成一个没有界面的电子表格，但是它可以处理相当大的数据集。Pandas 可以帮我们从 SQL 数据库、CSV（逗号分隔值文件）或者是其他类型的文件中读取数据、操作行或列、筛选数据以及某些情况下使用 matplotlib 进行数据可视化操作等。
+
+### Matplotlib
+
+Matplotlib 是一个用于绘制 2D 和 3D 数据的库。它对于图形、标签或者图层中的标记都有非常良好的支持。
+
+![通过梯度下降算法搜索一个价值函数的最优值，用 matplotlib 绘制的一对图形](/assets/2019/07/24/gradient-descent-cost-function-optimization.png)
+
+*通过梯度下降算法搜索一个价值函数的最优值，用 matplotlib 绘制的一对图形*
```
Comment (alick) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: cost function: 成本函数

File: _posts/2019-07-24-jupyter-and-data-science-in-fedora.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+$ pip3 install imblearn --user
+```
+
+### NLTK
+
+[自然语言工具箱](https://www.nltk.org)，或者叫 NLTK，可以帮我们处理人类语言数据，从而实现构建聊天机器人等工具（只是举个例子）。
+
+### SHAP
+
+机器学习算法非常擅长预测，但不善于解释他们为什么会得出该预测结果。[SHAP](https://github.com/slundberg/shap) 就是为了解决这个需求，方法就是分析训练的模型。
```
Comment (alick) [inline _posts/2019-07-24-jupyter-and-data-science-in-fedora.md]: 解决这个问题

**Review by alick — CHANGES_REQUESTED** [review]
小改

**Comment by alick** [comment]
“归一化”是对的。

**Comment by geekpany** [comment]
都改好了～

**Review by sztsian — APPROVED** [review]
